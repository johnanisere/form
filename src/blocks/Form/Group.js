import styled from "styled-components";
import Section from "../../elements/Section";

const Group = styled(Section)`
  flex-direction: column;
  padding: 20px;
`;

export default Group;
