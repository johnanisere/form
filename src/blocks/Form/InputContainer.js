import styled, { css } from "styled-components";
import Article from "../../elements/Article";

const InputContainer = styled(Article)`
  ${({ special }) =>
    special &&
    css`
      display: flex;
      justify-content: center;
      padding: 20px;
      input {
        font-size: 14px;
        color: rgb(255, 255, 255);
        outline: none;
        background: linear-gradient(rgb(99, 113, 199), rgb(85, 99, 193));
        &:hover ,&:focus{
          box-shadow: 2px 4px 11px 1px rgba(0,0,0,.2)
          transform: translateY(-1);
        }
      }
    `}
`;

export default InputContainer;
