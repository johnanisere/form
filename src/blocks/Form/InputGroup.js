import styled, { css } from "styled-components";
import P from "../../elements/P";

const InputGroup = styled(P)`
  ${({ special }) =>
    special &&
    css`
      display: flex;
      justify-content: center;
      padding: 20px;
      input {
        font-size: 14px;
        color: rgb(255, 255, 255);
        outline: none;
        background: linear-gradient(rgb(99, 113, 199), rgb(85, 99, 193));
        cursor: pointer;
      }
    `}
`;

export default InputGroup;
