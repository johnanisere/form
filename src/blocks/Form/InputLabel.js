import styled from "styled-components";
import Label from "../../elements/Label";

const InputLabel = styled(Label)`
  display: block;
  width: 100%;
  box-sizing: border-box;
  padding-bottom: 6px;
  color: #212b36;
  font-size: 14px;
`;

export default InputLabel;
