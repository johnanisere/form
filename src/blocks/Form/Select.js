import styled from "styled-components";
import MySelect from "../../elements/Select";

const Select = styled(MySelect)`
  width: 100%;
  display: block;
  box-sizing: border-box;
  border: 1px solid #e3e5e7;
  height: 120px;
  font-size: 14px;
  line-height: 1.42857143;
  padding: 18px;
  color: #555;
  background-color: #fff;
  background-image: none;
  border: 1px solid #ccc;
  border-radius: 4px;
  box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
  transition: border-color ease-in-out 0.15s, box-shadow ease-in-out 0.15s;
  &:focus {
    border-width: 2px !important;
    border-style: solid !important;
    border-color: rgb(92, 106, 196) !important;
    border-image: initial !important;
    outline: white;
  }
`;

export default Select;
