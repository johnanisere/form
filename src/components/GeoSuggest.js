import React from "react";
import Geosuggest from "react-geosuggest";

class App extends React.Component {
  onSuggestionSelect = suggestion => {
    if (suggestion) {
      const { name } = this.props;
      const data = {
        target: {
          name,
          value: suggestion.description
        }
      };
      this.props.onChange(data);
    }
  };

  render() {
    const { initialValue } = this.props;
    return (
      <Geosuggest
        placeholder="City"
        initialValue={initialValue}
        ref={el => (this._geoSuggest = el)}
        onSuggestSelect={this.onSuggestionSelect}
      />
    );
  }
}

export default App;
