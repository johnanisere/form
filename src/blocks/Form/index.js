import styled from "styled-components";

import Input from "./Input";
import Group from "./Group";
import Title from "./Title";
import Select from "./Select";
import InputGroup from "./InputGroup";
import InputLabel from "./InputLabel";
import MyForm from "../../elements/Form";
import InputContainer from "./InputContainer";

const Form = styled(MyForm)`
  width: 100%;
  max-width: 580px;
  box-shadow: rgba(63, 63, 68, 0.05) 0px 0px 0px 1px,
    rgba(63, 63, 68, 0.15) 0px 1px 3px 0px;
  border-radius: 3px;
  margin: 50px 0;

  .geosuggest__input {
    width: 100%;
    display: block;
    box-sizing: border-box;
    padding: 4px 18px;
    border: 1px solid #e3e5e7;
    height: 40px;
    font-size: 14px;
    line-height: 1.42857143;
    color: #555;
    background-color: #fff;
    background-image: none;
    border: 1px solid #ccc;
    border-radius: 4px;
    box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
    transition: border-color ease-in-out 0.15s, box-shadow ease-in-out 0.15s;
    &:focus {
      border-width: 2px !important;
      border-style: solid !important;
      border-color: rgb(92, 106, 196) !important;
      border-image: initial !important;
      outline: white;
    }
  }
  .geosuggest__suggests {
    list-style: none;
    padding: 0;
    background: #ffff;
    margin: 0;
    box-shadow: rgba(63, 63, 68, 0.05) 0px 0px 0px 1px,
      rgba(63, 63, 68, 0.15) 0px 1px 3px 0px;
    border-radius: 3px;
    li {
      padding: 0.5em 20px;
      cursor: pointer;
      &:hover {
        background: #f5f5f5;
      }
    }
  }
  .geosuggest__suggests--hidden {
    display: none;
  }
`;

Form.Title = Title;
Form.Input = Input;
Form.Group = Group;
Form.Select = Select;
Form.Label = InputLabel;
Form.InputGroup = InputGroup;
Form.InputContainer = InputContainer;

export default Form;
