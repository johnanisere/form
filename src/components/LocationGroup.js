import React from "react";

import Form from "../blocks/Form";
import GeoSuggest from "./GeoSuggest";

const LocationGroup = ({ title, required, name, onChange, value }) => (
  <Form.InputContainer>
    <Form.Label>
      <span>{title}</span>
      {required && (
        <strong>
          <abbr title="required">*</abbr>
        </strong>
      )}
    </Form.Label>
    <GeoSuggest name={name} onChange={onChange} initialValue={value} />
  </Form.InputContainer>
);

export default LocationGroup;
