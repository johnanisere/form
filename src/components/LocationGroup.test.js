import React from "react";
import { shallow } from "enzyme";

import LocationGroup from "./LocationGroup";
import Form from "../blocks/Form";
import GeoSuggest from "./GeoSuggest";

const props = {
  title: "",
  required: true,
  name: "",
  onChange: () => {},
  value: ""
};

it("LocationGroup component exists", () => {
  const wrapper = shallow(<LocationGroup {...props} />);
  expect(wrapper.exists()).toBe(true);
});

it("LocationGroup component renders", () => {
  const wrapper = shallow(<LocationGroup {...props} />);
  expect(wrapper).toMatchSnapshot();
});

it(`LocationGroup should contain "Form.Label", "Form.InputContainer","GeoSuggest" `, () => {
  const wrapper = shallow(<LocationGroup {...props} />);
  expect(wrapper.find(Form.Label)).toHaveLength(1);
  expect(wrapper.find(Form.InputContainer)).toHaveLength(1);
  expect(wrapper.find(GeoSuggest)).toHaveLength(1);
});
