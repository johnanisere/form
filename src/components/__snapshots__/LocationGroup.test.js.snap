// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationGroup component renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LocationGroup
    name=""
    onChange={[Function]}
    required={true}
    title=""
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef>
          <span>
            
          </span>
          <strong>
            <abbr
              title="required"
            >
              *
            </abbr>
          </strong>
        </ForwardRef>,
        <App
          initialValue=""
          name=""
          onChange={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <span>
              
            </span>,
            <strong>
              <abbr
                title="required"
              >
                *
              </abbr>
            </strong>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <abbr
                title="required"
              >
                *
              </abbr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "*",
                "title": "required",
              },
              "ref": null,
              "rendered": "*",
              "type": "abbr",
            },
            "type": "strong",
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-gZMcBi",
            "isStatic": true,
            "rules": Array [
              "
  display: block;
  width: 100%;
  box-sizing: border-box;
  padding-bottom: 6px;
  color: #212b36;
  font-size: 14px;
",
            ],
          },
          "displayName": "Styled(styled.label)",
          "foldedComponentIds": Array [
            "sc-iwsKbI",
          ],
          "render": [Function],
          "styledComponentId": "sc-gZMcBi",
          "target": "label",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "initialValue": "",
          "name": "",
          "onChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-jTzLTM",
        "isStatic": false,
        "rules": Array [
          "
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "Styled(styled.article)",
      "foldedComponentIds": Array [
        "sc-VigVT",
      ],
      "render": [Function],
      "styledComponentId": "sc-jTzLTM",
      "target": "article",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            <span>
              
            </span>
            <strong>
              <abbr
                title="required"
              >
                *
              </abbr>
            </strong>
          </ForwardRef>,
          <App
            initialValue=""
            name=""
            onChange={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <span>
                
              </span>,
              <strong>
                <abbr
                  title="required"
                >
                  *
                </abbr>
              </strong>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <abbr
                  title="required"
                >
                  *
                </abbr>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "*",
                  "title": "required",
                },
                "ref": null,
                "rendered": "*",
                "type": "abbr",
              },
              "type": "strong",
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-gZMcBi",
              "isStatic": true,
              "rules": Array [
                "
  display: block;
  width: 100%;
  box-sizing: border-box;
  padding-bottom: 6px;
  color: #212b36;
  font-size: 14px;
",
              ],
            },
            "displayName": "Styled(styled.label)",
            "foldedComponentIds": Array [
              "sc-iwsKbI",
            ],
            "render": [Function],
            "styledComponentId": "sc-gZMcBi",
            "target": "label",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "initialValue": "",
            "name": "",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-jTzLTM",
          "isStatic": false,
          "rules": Array [
            "
  ",
            [Function],
            "
",
          ],
        },
        "displayName": "Styled(styled.article)",
        "foldedComponentIds": Array [
          "sc-VigVT",
        ],
        "render": [Function],
        "styledComponentId": "sc-jTzLTM",
        "target": "article",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
